charts/ccsm-helm/test/operate/deployment_test.go

```diff
@@ -0,0 +1,281 @@
+	var deployment appsv1.Deployment
+	helm.UnmarshalK8SYaml(s.T(), output, &deployment)
+
+	// then
+	expectedContainerImage := "camunda/operate:a.b.c"
+	containers := deployment.Spec.Template.Spec.Containers
+	s.Require().Equal(1, len(containers))
+	s.Require().Equal(expectedContainerImage, containers[0].Image)
+}
+
+func (s *deploymentTemplateTest) TestContainerOverwriteImageTagWithChartDirectSetting() {
+	// given
+	options := &helm.Options{
+		SetValues: map[string]string{
+			"global.image.tag":  "x.y.z",
+			"operate.image.tag": "a.b.c",
+		},
+		KubectlOptions: k8s.NewKubectlOptions("", "", s.namespace),
+		ExtraArgs:      map[string][]string{"template": {"--debug"}, "install": {"--debug"}},
+	}
+
+	// when
+	output := helm.RenderTemplate(s.T(), options, s.chartPath, s.release, s.templates)
+	var deployment appsv1.Deployment
+	helm.UnmarshalK8SYaml(s.T(), output, &deployment)
+
+	// then
+	expectedContainerImage := "camunda/operate:a.b.c"
+	containers := deployment.Spec.Template.Spec.Containers
+	s.Require().Equal(1, len(containers))
+	s.Require().Equal(expectedContainerImage, containers[0].Image)
+}
+
+func (s *deploymentTemplateTest) TestContainerSetContainerCommand() {
+	// given
+	options := &helm.Options{
+		SetValues: map[string]string{
+			"operate.command": "[printenv]",
+		},
+		KubectlOptions: k8s.NewKubectlOptions("", "", s.namespace),
+		ExtraArgs:      map[string][]string{"template": {"--debug"}, "install": {"--debug"}},
+	}
+
+	// when
+	output := helm.RenderTemplate(s.T(), options, s.chartPath, s.release, s.templates)
+	var deployment appsv1.Deployment
+	helm.UnmarshalK8SYaml(s.T(), output, &deployment)
+
+	// then
+	containers := deployment.Spec.Template.Spec.Containers
+	s.Require().Equal(len(containers), 1)
+	s.Require().Equal(1, len(containers[0].Command))
+	s.Require().Equal("printenv", containers[0].Command[0])
+}
+
+func (s *deploymentTemplateTest) TestContainerSetExtraVolumes() {
+	// given
+	options := &helm.Options{
+		SetValues: map[string]string{
+			"operate.extraVolumes[0].name":                  "extraVolume",
+			"operate.extraVolumes[0].configMap.name":        "otherConfigMap",
+			"operate.extraVolumes[0].configMap.defaultMode": "744",
+		},
+		KubectlOptions: k8s.NewKubectlOptions("", "", s.namespace),
+		ExtraArgs:      map[string][]string{"template": {"--debug"}, "install": {"--debug"}},
+	}
+
+	// when
+	output := helm.RenderTemplate(s.T(), options, s.chartPath, s.release, s.templates)
+	var deployment appsv1.Deployment
+	helm.UnmarshalK8SYaml(s.T(), output, &deployment)
+
+	// then
+	volumes := deployment.Spec.Template.Spec.Volumes
+	s.Require().Equal(len(volumes), 2)
+
+	extraVolume := volumes[1]
+	s.Require().Equal("extraVolume", extraVolume.Name)
+	s.Require().NotNil(*extraVolume.ConfigMap)
+	s.Require().Equal("otherConfigMap", extraVolume.ConfigMap.Name)
+	s.Require().Equal(int32(744), *extraVolume.ConfigMap.DefaultMode)
```

Comment:
:wrench: I think you can use `s.Require().EqualValues` here, which will make some smart type conversion. See https://pkg.go.dev/github.com/stretchr/testify/assert?utm_source=godoc#EqualValues

Reply:
This is nice :rocket: